File: web/WebFirstLab/WithAjax/php/RadioContainer.php
<?php
    require_once('Containable.php');
    require_once('WrongInputException.php');

    class RadioContainer implements Containable {
        private $containable;
        private $array;

        public function __construct(Containable $containable, array $array) {
            $this->containable = $containable;
            $this->array = $array;
        }

        public function content() {
            $content = $this->containable->content();

            if(in_array($content, $this->array)) {
                return $content;
            }

            throw new WrongInputException();
        }
    }
?>

File: web/WebSecondLab/src/main/java/app/entities/classes/RadioContainer.java
package app.entities.classes;

import app.entities.exceptions.WrongInputException;
import app.entities.interfaces.Containable;

import java.util.List;

public class RadioContainer implements Containable<Float> {
    private final Containable<Float> containable;
    private final List<Float> options;

    public RadioContainer(Containable<Float> containable, List<Float> options) {
        this.containable = containable;
        this.options = options;
    }

    @Override
    public Float content() throws WrongInputException {
        float content = containable.content();

        if (options.contains(content)) {
            return content;
        }

        throw new WrongInputException("There is no such option");
    }
}
